sonar-scanner-engine/src/main/java/org/sonar/batch/cpd/CpdExecutor.java

```diff
@@ -206,7 +205,7 @@ private Duplication toReportDuplication(BatchComponent component, Duplication.Bu
           blockBuilder.setOtherFileRef(sameProjectComponent.batchId());
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'sameProjectComponent' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2259)